dm/pkg/checker/privilege_test.go

```diff
@@ -334,3 +334,67 @@ func TestVerifyReplicationPrivileges(t *testing.T) {
 		}
 	}
 }
+
+func TestVerifyTargetPrivilege(t *testing.T) {
+	cases := []struct {
+		grants     []string
+		checkState State
+		errStr     string
+	}{
+		{
+			grants:     nil, // non grants
+			checkState: StateFailure,
+			errStr:     "there is no such grant defined for current user on host '%'",
+		},
+		{
+			grants:     []string{"invalid SQL statement"},
+			checkState: StateFailure,
+			errStr:     "line 1 column 7 near \"invalid SQL statement\" ",
+		},
+		{
+			grants:     []string{"CREATE DATABASE db1"}, // non GRANT statement
+			checkState: StateFailure,
+			errStr:     "CREATE DATABASE db1 is not grant statement",
+		},
+		{
+			grants: []string{
+				"GRANT ALL PRIVILEGES ON *.* TO 'user'@'%'",
+			},
+			checkState: StateSuccess,
+		},
+		{
+			grants: []string{
+				"GRANT SELECT, CREATE, INSERT, UPDATE, DELETE, ALTER, DROP ON *.* TO 'root'@'%'",
+			},
+			checkState: StateSuccess,
+		},
+		{
+			grants: []string{
+				"GRANT SELECT, INSERT, DELETE, ALTER, DROP ON *.* TO 'root'@'%'",
+			},
+			checkState: StateFailure,
```

Comment:
Since we can't be precise enough on what table the binlog will write, we should tend to be too loose rather than too strict.

Reply:
What about reporting a warning instead of a failure

Reply:
edf4739